net/packet/packet.go

```diff
@@ -55,27 +55,44 @@ func (p *Packet) Pack(w io.Writer, threshold int) error {
 func (p *Packet) packWithoutCompression(w io.Writer) error {
 	buffer := bufPool.Get().(*bytes.Buffer)
 	defer bufPool.Put(buffer)
-	buffer.Reset()
-	n, err := VarInt(p.ID).WriteTo(buffer)
-	if err != nil {
-		panic(err)
-	}
-	// Length
-	_, err = VarInt(int(n) + len(p.Data)).WriteTo(w)
-	if err != nil {
-		return err
+
+	if buffer.Len() < 3 {
+		var padding [3]byte
+		buffer.Write(padding[:3-buffer.Len()])
```

Comment:
1. You can consider use `.Grow()` here.
2. It's wired to only grow to only have a size of 3. You write entire `p.Data` later, 3 bytes will not enough.

Reply:
The 3 bytes are to reserve room for the length header at the front of the packet. If I used Grow(), the capacity would be increased, not the length. Basically, the order is [Three bytes][Packet ID][Packet Data]. Then, measure the length of ID + data and write that to the first three bytes.

Reply:
> Does the examples still work after this modification?

I will test the examples momentarily

Reply:
I can verify that `simpleServer1.17.1`, `pressureTest`, `minimal`, `mcping`, and `daze` all work correctly. I also added 41c7440 which adds a comment about the 3-byte reservation because I feel like it could be explained a bit more clearly. Does that make sense?

Reply:
I feel this `if` is unnecessary. It can be replaced by `buffer.Reset()` following `buffer.Write([]byte{0, 0, 0})`

Reply:
I suppose that's true. I made that change in the latest commit.